src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -46,50 +48,74 @@ private void CompilationStartAction(CompilationStartAnalysisContext context)
             var idToCachedResult = new ConcurrentDictionary<Guid, ConcurrentDictionary<string, string>>(
                 concurrencyLevel: 2, capacity: 0);
 
-            context.RegisterSymbolAction(c => SymbolAction(c, idToCachedResult), _symbolKinds);
+            context.RegisterSymbolAction(SymbolAction, _symbolKinds);
+
+            void SymbolAction(SymbolAnalysisContext symbolContext)
```

Comment:
"// local functions" comment? :-/ that seems to me like "warning, new language features!"

Reply:
Yes.  Because local functions require extra attention over a normal method.  Especially related to understanding capture flow.  There is no obvious visual distinction between them and methods, so having the comment is very helpful.  Otherwise, *especially* in github, you have to scroll all the way up to figure out if you are preceded by a statement and thus are a local function vs a sibling method.

Similarly, groupign all the local functions together, and not interspersing them in code, makes things much easier to analyze as you can look at the actual logic and code-flow in a single pass, without having random declarations interspersed that you have to jump past.

Like with *all* language constructs, there are ways to use them that can aide readability and comprehension and ways to use them that can hinder them.  This is just like informal rules the IDE has about complexity in expressions and other things.  You are not disallowed from using these features, but please use them in manners that are most conducive to a team environment.

Reply:
fixed